docs/source/traitsui_user_manual/testing.rst

```diff
@@ -4,656 +4,54 @@
-
-    tester = UITester(registries=[custom_registry])
-
-If we have also added a custom ``ManyMouseClick`` interaction (see section
-above), we can write test code like this::
-
-    container = UITester().find_by_id(ui, "some_container")
-    button_wrapper = container.locate(NamedButton("OK"))
-    button_wrapper.perform(ManyMouseClick(n_times=10))
-
-where both ``NamedButton`` and ``ManyMouseClick`` are custom objects.
-
-Overriding TraitsUI Builtin Testing Support
--------------------------------------------
-
-|UITester| maintains a list of registries ordered in decreasing priority.
-For example, if you provide multiple registries::
-
-    tester = UITester(registries=[custom_registry, another_registry])
+How-to guides
+=============
 
-Interactions and locations registered in the first registry will supersede that
-of the second (if such implementation exists). Builtin support for TraitsUI
-editors is added last, hence with the lowest priority.
+.. toctree::
+   :maxdepth: 1
 
-With that, one can override TraitsUI builtin testing support by redefining the
-interaction handler and/or location solver via an instance of
-|TargetRegistry|::
+   testing/howtos/add_new_interaction
+   testing/howtos/add_new_location
 
-    from traitsui.qt4.button_editor import SimpleEditor
-    from traitsui.testing.api import command
 
-    custom_registry = TargetRegistry(
-        target_class=SimpleEditor,
-        locator_class=command.MouseClick,
-        handler=my_custom_handler,
-    )
-    tester = UITester(registries=[custom_registry])
+Discussions
+===========
 
-..
-   # substitutions
+.. toctree::
+   :maxdepth: 1
 
-.. |testing.api| replace:: :mod:`~traitsui.testing.api`
+   testing/discussions/automated_vs_manual_test
+   testing/discussions/compatibility_pyface_testing
+   testing/discussions/target_interaction_location
+   testing/discussions/working_of_extensions
 
-.. |Group| replace:: :class:`~traitsui.group.Group`
-.. |Item| replace:: :class:`~traitsui.item.Item`
-.. |UI| replace:: :class:`~traitsui.ui.UI`
+Technical References
+====================
 
-.. |MouseClick| replace:: :class:`~traitsui.testing.tester.command.MouseClick`
-.. |KeySequence| replace:: :class:`~traitsui.testing.tester.command.KeySequence`
-.. |DisplayedText| replace:: :class:`~traitsui.testing.tester.query.DisplayedText`
-.. |TargetById| replace:: :class:`~traitsui.testing.tester.locator.TargetById`
-.. |TargetByName| replace:: :class:`~traitsui.testing.tester.locator.TargetByName`
+.. toctree::
+   :maxdepth: 1
 
-.. |TargetRegistry| replace:: :class:`~traitsui.testing.tester.target_registry.TargetRegistry`
-.. |TargetRegistry.register_interaction| replace:: :func:`~traitsui.testing.tester.target_registry.TargetRegistry.register_interaction`
-.. |TargetRegistry.register_location| replace:: :class:`~traitsui.testing.tester.target_registry.TargetRegistry.register_location`
+   testing/references/examples
+   API references <../api/traitsui.testing>
```

Comment:
nit: I believe in other Ets documentation we just say "API Reference" rather than references